### PR TITLE
contract.events.X.getLogs() support without creating filters

### PR DESCRIPTION
**Pull request by miohtama**
### What was wrong?

Related to Issue #1177

### How was it fixed?

Added a proposed implementation.

Still working on the polish and documentation and test coverage. Opened a PR for early feedback.

#### Cute Animal Picture

```
\|/          (__)    
     `\------(oo)
       ||    (__)
       ||w--||     \|/
   \|/

```

### CODE REVIEW

File: web3/contract.py [hunk 1]
```diff
@@ -1323,6 +1350,72 @@ def build_filter(self):
...
+            toBlock=toBlock,
+            address=self.address,
+        )
+
+        # Call JSON-RPC API
+        logs = self.web3.eth.getLogs(event_filter_params)
+
+        # Convert raw binary data to Python proxy objects as described by ABI
+        for entry in logs:
+            yield get_event_data(abi, entry)
```
Comment (pipermerriam) [inline web3/contract.py]: I think using a `yield` pattern here is likely to cause confusion.  Maybe go ahead and decorate this function using `to_tuple` or turn this into  `return tuple(get_event_data(abi, entry) for entry in logs)`

Comment (miohtama) [reply web3/contract.py]: Alrighty. We are not streaming the response from the node any case, so there is little marginal gain here.

Comment (miohtama) [reply web3/contract.py]: Fixed in the latest revision.

**Comment by miohtama** [comment]
Ready to merge if no further comments  in a new review.

File: tests/core/filtering/test_contract_getLogs.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    txn_hashes.append(
+        emitter.functions.logTriple(event_id, 1, 3, 1).transact()
+    )
+    for txn_hash in txn_hashes:
+        wait_for_transaction(web3, txn_hash)
+
+    all_logs = contract.events.LogTripleWithIndex.getLogs()
+    assert len(all_logs) == 4
+
+    # Filter all entries where arg2 in (1, 2)
```
Comment (kclowes) [inline tests/core/filtering/test_contract_getLogs.py]: Nitpick

I think this comment is wrong. Should be `arg1`?

File: web3/contract.py [hunk 3]
```diff
@@ -1323,6 +1350,92 @@ def build_filter(self):
...
+                 'blockHash': HexBytes('...'),
+                 'blockNumber': 3
+                }),
+                AttributeDict(...),
+                ...
+            )
+
+        See also: :func:`web3.middleware.filter.local_filter_middleware`.
+
+        :param argument_filters: TODO
```
Comment (kclowes) [inline web3/contract.py]: Nitpick

Did you want to fill this in? If not, let's get rid of the `TODO` :)

**Pull request merged**